danger/src/PullRequestManager.ts

```diff
@@ -0,0 +1,165 @@
+import { GitHubPRDSL } from 'danger/distribution/dsl/GitHubDSL';
+
+import { Octokit } from '@octokit/rest';
+import { GithubApiWrapper } from './GithubApiWrapper';
+import { getPackageChangelogRelativePath } from './Utils';
+
+export enum ChangelogEntryType {
+  BUG_FIXES = 'bug-fix',
+  NEW_FEATURES = 'new-feature',
+  BREAKING_CHANGES = 'breaking-change',
+}
+
+export const DEFAULT_ENTRY_TYPE = ChangelogEntryType.BUG_FIXES;
+
+export type ChangelogEntry = {
+  type: ChangelogEntryType;
+  message: string;
+};
+
+export const DEFAULT_CHANGELOG_ENTRY_KEY = 'default' as const;
+
+export type ChangelogEntries = {
+  [DEFAULT_CHANGELOG_ENTRY_KEY]: ChangelogEntry;
+  [key: string]: ChangelogEntry;
+};
+
+export type PullRequest = GitHubPRDSL | Octokit.PullsListResponseItem;
+
+const dangerMessage = `Add missing changelog`;
+const dangerTags = `[danger][bot]`;
+
+export class PullRequestManager {
+  constructor(private pullRequest: PullRequest, private githubApi: GithubApiWrapper) {}
+
+  /**
+   * Get suggested changelog entries from PR provided in the constructor.
+   *
+   * If PR doesn't contais `# Changelog` section, this method returns:
+   * {
+   *   [DEFAULT_CHANGELOG_ENTRY_KEY]: <title of this pr without tags>
+   * }
+   * Otherwise, it tries to parse PR's body.
+   */
+  parseChangelogSuggestionFromDescription(): ChangelogEntries {
+    const changelogEntries = {
+      [DEFAULT_CHANGELOG_ENTRY_KEY]: {
+        type: DEFAULT_ENTRY_TYPE,
+        message: this.pullRequest.title.replace(/\[.*\]/, '').trim(),
+      },
+    };
+
+    const changelogTag = this.pullRequest.body
+      .match(/#\schangelog(([^#]*?)\s?)*/i)?.[0]
+      ?.replace(/^-/, '');
+    if (changelogTag) {
+      changelogTag
+        .split('\n')
+        .slice(1)
+        .map(line => line.trim())
+        .filter(line => line.length > 0)
+        .forEach(line => {
+          const tags = this.parseTagsFromLine(line);
+          if (!tags) {
+            warn(`Couldn't parse line: ${line}.`);
+            return;
+          }
+
+          changelogEntries[tags.packageName] = {
+            type: tags.type,
+            message: line.replace(/\[.*\]/, '').trim(),
+          };
+        });
+    }
```

Comment:
Probably having expotools API exposed (as I suggested in another comment) would make it simpler because we could tokenize this as markdown 🤔